core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -615,11 +615,13 @@ public function create($set=array())
 			throw new InternalServerErrorException('Table "' . $this->strTable . '" is not creatable.');
 		}
 
+		$databaseFields = array_column($this->Database->listFields($this->strTable), 'name');
+
 		// Get all default values for the new entry
 		foreach ($GLOBALS['TL_DCA'][$this->strTable]['fields'] as $k=>$v)
 		{
 			// Use array_key_exists here (see #5252)
-			if (\array_key_exists('default', $v))
+			if (\array_key_exists('default', $v) && \in_array($k, $databaseFields, true))
```

Comment:
Do we need to compare the columns in lowercase?

Reply:
Yes, `$this->Database->listFields($this->strTable)` will return the column names in lower case.

Reply:
I don't think so. Doctrine does, but our method does not.

Reply:
Would the case match if you use entities as well?

I’d prefer to compare them in lowercase to be sure.

Reply:
Yeah, you are right - our method does not return it in lower case, my bad.

Reply:
Fact is that most DCAs use lower-case keys, so for this to work, the columns names must be lower-case, too, right?

Reply:
We would need to lowercase them both, `$databaseFields` as well as `$k`, to be sure that we compare correctly IMO.

Reply:
I have no idea what you're talking about. The DCA uses the exact same column names as the database, we never lowercase that anywhere. Or should we lowercase the field names when writing to the database as well???